magic_spoiler/__main__.py

```diff
@@ -23,9 +23,23 @@
 OUTPUT_DIR = pathlib.Path("out")
 OUTPUT_TMP_DIR = OUTPUT_DIR.joinpath("tmp")
 XML_ESCAPE_TRANSLATE_MAP = str.maketrans(
-    {"&": "&amp;", '"': "&quot;", "<": "&lt;", ">": "&gt;",
-    # remove any control characters outright
-    **{chr(i): "" for i in range(ord(" "))}}
+    {"&": "&amp;", '"': "&quot;", "<": "&lt;", ">": "&gt;"}
+)
+# remove any control characters outright
+XML_ESCAPE_TRANSLATE_MAP.update({i: "" for i in range(ord(" "))})
+# don't remove whitespace characters in the sub " " range
+del XML_ESCAPE_TRANSLATE_MAP[ord("\n")]
+del XML_ESCAPE_TRANSLATE_MAP[ord("\t")]
+
+# copied from Cockatrice/oracle/src/oracleimporter.h OracleImporter::mainCardTypes
+MAINTYPES = (
+    "Planeswalker",
+    "Creature",
+    "Land",
+    "Sorcery",
+    "Instant",
+    "Artifact",
+    "Enchantment"
```

Comment:
Would it be helpful to get this from [MTGJSON CardTypes JSON file](https://mtgjson.com/downloads/all-files/#cardtypes) instead having to copy and paste from Oracle source code, and make sure it stays up to date?

Reply:
this list's order is very important, it has to be this way in order to select the main cardtype the way we do in oracle.